pkg/controllermanager/controller/project/project_control.go

```diff
@@ -68,80 +70,58 @@ func (c *Controller) projectDelete(obj interface{}) {
-		return err
-	}
-
-	project, err := c.projectLister.Get(name)
-	if apierrors.IsNotFound(err) {
-		logger.Logger.Debugf("[PROJECT RECONCILE] %s - skipping because Project has been deleted", key)
-		return nil
-	}
-	if err != nil {
-		logger.Logger.Infof("[PROJECT RECONCILE] %s - unable to retrieve object from store: %v", key, err)
-		return err
-	}
-
-	needsRequeue, err := c.control.ReconcileProject(project)
-	if err != nil {
-		return err
+// NewProjectReconciler creates a new instance of a reconciler which reconciles Projects.
+func NewProjectReconciler(l logrus.FieldLogger, config config.ControllerManagerControllerConfiguration, clientMap clientmap.ClientMap, k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory, recorder record.EventRecorder, namespaceLister kubecorev1listers.NamespaceLister) reconcile.Reconciler {
+	return &projectReconciler{
+		logger:                 l,
+		config:                 config,
+		clientMap:              clientMap,
+		k8sGardenCoreInformers: k8sGardenCoreInformers,
+		recorder:               recorder,
+		namespaceLister:        namespaceLister,
 	}
-	if needsRequeue {
-		c.projectQueue.AddAfter(key, time.Minute)
-	}
-
-	return nil
-}
-
-// ControlInterface implements the control logic for updating Projects. It is implemented as an interface to allow
-// for extensions that provide different semantics. Currently, there is only one implementation.
-type ControlInterface interface {
-	// ReconcileProject implements the control logic for Project creation, update, and deletion.
-	// If an implementation returns a non-nil error, the invocation will be retried using a rate-limited strategy.
-	// Implementors should sink any errors that they do not wish to trigger a retry, and they may feel free to
-	// exit exceptionally at any point provided they wish the update to be re-run at a later point in time.
-	ReconcileProject(project *gardencorev1beta1.Project) (bool, error)
 }
 
-// NewDefaultControl returns a new instance of the default implementation ControlInterface that
-// implements the documented semantics for Projects. updater is the UpdaterInterface used
-// to update the status of Projects. You should use an instance returned from NewDefaultControl() for any
-// scenario other than testing.
-func NewDefaultControl(clientMap clientmap.ClientMap, config *config.ControllerManagerConfiguration, k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory, recorder record.EventRecorder, namespaceLister kubecorev1listers.NamespaceLister) ControlInterface {
-	return &defaultControl{clientMap, config, k8sGardenCoreInformers, recorder, namespaceLister}
-}
-
-type defaultControl struct {
+type projectReconciler struct {
+	logger                 logrus.FieldLogger
+	config                 config.ControllerManagerControllerConfiguration
 	clientMap              clientmap.ClientMap
-	config                 *config.ControllerManagerConfiguration
 	k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory
```

Comment:
I think the `k8sGardenCoreInformers` can be removed as they aren't used anywhere.